File: Two Pointer/334. Increasing Triplet Subsequence.py
class Solution(object):
    def increasingTriplet(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        if not nums or len(nums) < 3:
            return False
        small, large = sys.maxint, sys.maxint
        for num in nums:
            if num < small:
                small = num
            elif small < num < large:
                large = num
            elif num > large:
                return True
        return False

File: DP/334. Increasing Triplet Subsequence.cpp
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        vector<int> dp;
        for(auto& num:nums)
        {
            if(dp.size() == 0)
                dp.push_back(num);
            else if(num <= dp[0])
            {
                dp[0] = num;
            }
            else if(dp.size() == 1)
            {
                dp.push_back(num);
            }
            else if(num <= dp[1])
                dp[1] = num;
            else if(num > dp[1])
                return true;
        }
        return false;
    }
};
